src/app.js

```diff
@@ -7,6 +7,17 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  element = root.firstChild
+  node = hydrate(element)
```

Comment:
Does this need to be a function?  Why couldn't it just be an `if`?  Something like:

```js
if (element === null) {
  node = {
    tag: element.tagName,
    data: {},
    children: [].map.call(elm.childNodes, function(child){hydrate(child)}
  }
}